### DISCUSSION TITLE
Supported dataTypes

### DISCUSSION THREAD

**Discussion by seeser** [zinggAI/zingg#197] [General]
Are dataTypes other than string supported? Is there a list of dataType names? I only see "string" in the sample configs.

**Comment by sonalgoyal** [comment]
yes we support INTEGER, DOUBLE, FLOAT and DATE for matching(https://github.com/zinggAI/zingg/blob/main/core/src/main/java/zingg/feature/FeatureFactory.java). If you just want to pass through some types without matching them, they should work. We are working on exposing the type interface so that user defined types can also be added.  

Happy to help if you want to add specific types.

**Reply by sonalgoyal** [reply]
https://github.com/zinggAI/zingg/issues?q=is%3Aissue+integer
